xcode/Subconscious/Shared/Components/Common/Profile/DidQrCodeView.swift

```diff
@@ -36,24 +37,80 @@ struct DidQrCodeView: View {
         let col = CIColor(red: r, green: g, blue: b, alpha: a)
         maskFilter.inputImage = CIImage(color: col)
         
-        guard let img = maskFilter.outputImage else { return nil }
-        guard let cgImg = context.createCGImage(img, from: img.extent) else { return nil }
+        guard let img = maskFilter.outputImage else {
+            return fallback
+        }
+        
+        let scaleFactor = 10.0
+        let upscaleImg = img
+            .samplingNearest()
+            .transformed(
+                by: CGAffineTransformMakeScale(scaleFactor, scaleFactor)
+            )
```

Comment:
Couple hoops to jump through to ensure we get sharp pixels (nearest-neighbour magnification filter)

Reply:
Great stuff